common/services/prismic/utils.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import moment from 'moment';
+import {Predicates} from 'prismic-javascript';
+import {london} from '../../utils/format-date';
+import {getNextWeekendDateRange} from '../../utils/dates';
+import type {Period} from '../../model/periods';
+
+export function getPeriodPredicates(
+  period: ?Period,
+  startField: string,
+  endField: string
+): Predicates[] {
+  const now = london(new Date());
+  const startOfDay = moment().startOf('day');
+  const endOfDay = moment().endOf('day');
+  const weekendDateRange = getNextWeekendDateRange(now);
+
+  const predicates =
+    period === 'coming-up' ? [
+      Predicates.dateAfter(startField, startOfDay.toDate())
+    ] : period === 'current-and-coming-up' ? [
+      Predicates.dateAfter(endField, endOfDay.toDate())
+    ] : period === 'past' ? [
+      Predicates.dateBefore(endField, startOfDay.toDate())
+    ] : period === 'today' ? [
+      Predicates.dateBefore(startField, endOfDay.toDate()),
+      Predicates.dateAfter(endField, startOfDay.toDate())
+    ] : period === 'this-weekend' ? [
+      Predicates.dateBefore(startField, weekendDateRange.end),
+      Predicates.dateAfter(endField, weekendDateRange.start)
+    ] : period === 'this-week' ? [
```

Comment:
Do we use this, or do we only use the next seven days?

Reply:
We've called it `this-week` in `getExhibitionAndEventPromos`, but that's probably wrong, but it does what it says here.

When I deprecate `getExhibitionAndEventPromos` from the homepage, we'll add a new one here, which is `next-seven-days`.

Reply:
jammin'